lib/base/sysloglogger.cpp

```diff
@@ -40,6 +68,23 @@ void SyslogLogger::StatsFunc(const Dictionary::Ptr& status, const Array::Ptr&)
 	status->Set("sysloglogger", nodes);
 }
 
+void SyslogLogger::ValidateFacility(const String& value, const ValidationUtils& utils)
+{
+	ObjectImpl<SyslogLogger>::ValidateFacility(value, utils);
+
+	std::map<String, int>::const_iterator iterator = m_FacilityMap.find(value);
+
+	if (iterator != m_FacilityMap.end()) {
+		m_Facility = iterator->second;
+	} else {
+		try {
+			m_Facility = boost::lexical_cast<int>(value);
+		} catch(...) {
+			BOOST_THROW_EXCEPTION(ValidationError(this, boost::assign::list_of("facility"), "No matching facility provided."));
+		}
+	}
+}
```

Comment:
The validation method must not set a class variable. It should only throw an error in case a wrong value was given.

Instead, initialization must happen inside the class constructor.

Look into the db_ido categories for a similar implementation.

https://github.com/Icinga/icinga2/blob/master/lib/db_ido/dbconnection.cpp#L456